File: src/main/java/net/masterthought/dlanguage/stubs/DLanguageImportDeclarationStub.java
package net.masterthought.dlanguage.stubs;

import com.intellij.psi.stubs.IStubElementType;
import com.intellij.psi.stubs.NamedStubBase;
import com.intellij.psi.stubs.StubElement;
import com.intellij.util.io.StringRef;
import net.masterthought.dlanguage.psi.DLanguageImport;

public class DLanguageImportDeclarationStub extends NamedStubBase<DLanguageImport> {
    public DLanguageImportDeclarationStub(StubElement parent, IStubElementType elementType, StringRef name) {
        super(parent, elementType, name);
    }

    public DLanguageImportDeclarationStub(StubElement parent, IStubElementType elementType, String name) {
        super(parent, elementType, name);
    }
}

File: dlang/psi-api/src/main/kotlin/io/github/intellij/dlanguage/stubs/DLanguageImportDeclarationStub.kt
package io.github.intellij.dlanguage.stubs

import com.intellij.psi.stubs.IStubElementType
import com.intellij.psi.stubs.StubBase
import com.intellij.psi.stubs.StubElement
import io.github.intellij.dlanguage.psi.DLanguageImportDeclaration
import io.github.intellij.dlanguage.psi.interfaces.DVisibilityStubKind

class DLanguageImportDeclarationStub(
    val parent: StubElement<*>,
    val elementType: IStubElementType<*, *>,
    val visibility: DVisibilityStubKind,
    val packageVisibilityName: String?) : StubBase<DLanguageImportDeclaration>(parent, elementType) {
}
